tagformat/caseconversion/case_conversion.go

```diff
@@ -206,6 +206,31 @@ func DecodeUpperSnakeCase(s string) (DecodedIdentifier, error) {
 	return words, nil
 }
 
+// DecodeCasePreservingSnakeCase decodes Case_Preserving_Snake_Case into a
+// slice of lower-cased sub-string
+func DecodeCasePreservingSnakeCase(s string) (DecodedIdentifier, error) {
+	if unicode.IsDigit(rune(s[0])) {
```

Comment:
Hmm, I like the check, (and I see it's the same as some of the other decoding functions here), but by only using index `0`, runes that encode using multiple bytes in utf8 won't be interpreted correctly. (I must have missed that in Kevin's PRs)

I think you want to call [`utf8.DecodeRuneInString()`](https://golang.org/pkg/unicode/utf8/#DecodeRuneInString) here.

Alternatively, (and this is a lot less efficient) you can convert `s` to a `[]rune` and then convert it back to a `string` when you return.

Reply:
Good catch, changed it for all the cases relying on that